pkg/api/login.go

```diff
@@ -0,0 +1,82 @@
+package api
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/virtual-vgo/vvgo/pkg/login"
+	"github.com/virtual-vgo/vvgo/pkg/tracing"
+	"net/http"
+	"time"
+)
+
+// PasswordLoginHandler authenticates requests using form values user and pass and a static map of valid combinations.
+// If the user pass combo exists in the map, then a login cookie with the mapped roles is create and sent in the response.
+type PasswordLoginHandler struct {
+	Sessions *login.Store
+
+	// Logins is a map of login user and pass to a slice of roles for that login.
+	Logins map[[2]string][]login.Role
+}
+
+func (x PasswordLoginHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, span := tracing.StartSpan(r.Context(), "password_login")
+	defer span.Send()
+
+	if r.Method != http.MethodPost {
+		methodNotAllowed(w)
+		return
+	}
+
+	var identity login.Identity
+	if err := x.Sessions.ReadSessionFromRequest(ctx, r, &identity); err == nil {
+		http.Redirect(w, r, "/", http.StatusFound)
+		return
+	}
+
+	user := r.FormValue("user")
+	pass := r.FormValue("pass")
+
+	gotRoles, ok := x.Logins[[2]string{user, pass}]
+	if !ok {
+		logger.WithFields(logrus.Fields{
+			"user": user,
+		}).Error("password authentication failed")
+		unauthorized(w)
+		return
+	}
+
+	identity = login.Identity{
+		Kind:  login.KindPassword,
+		Roles: gotRoles,
+	}
+
+	cookie, err := x.Sessions.NewCookie(ctx, &identity, 31557600*time.Second)
```

Comment:
Where does `31557600` come from?

Reply:
Oh ya I think that's a years worth of seconds haha. This should probably be an env var or something.